### DISCUSSION TITLE
Inject header "Access-Control-Allow-Origin":  "http://localhost:8000" on all responses

### DISCUSSION THREAD

**Discussion by paolooo** [bbyars/mountebank#743] [Q&A]
Hi,

How can we inject the header `{ "Access-Control-Allow-Origin":  "http://localhost:8000" }` programmatically on all stub responses?

Currently, what I'm doing is:

```json
{
  // ...
  "responses": [
    {
      "is": {
        "headers": {
          "Access-Control-Allow-Origin": "http://localhost:8000",
        },
      },
      // ...
    },
    {
      "is": {
        "headers": {
          "Access-Control-Allow-Origin": "http://localhost:8000",
        },
      },
      // ...
    },
  ]
}
```

Is it possible that we can have an injection option in **Imposter Creation parameters** just like with `tcp` imposter that has `endOfRequestResolver` option.

Thanks.

**Comment by bbyars** [comment]
Hi Paolo,
Have you tried using the defaultResponse
<https://www.mbtest.org/docs/protocols/http> at the imposter level to set
it? As long as it's static and doesn't require dynamic information, I
believe that's the easiest option.
-Brandon

**Reply by paolooo** [reply]
Hi Brandon,

Yes, I've tested `defaultResponse` I thought this was the answer too. It works only if the stub response doesn't have a `header`. For example,

```
"defaultResponse": {
  "headers": {
      "Access-Control-Allow-Origin": "http://localhost:8000"
  }
},
"stubs": [
  {
    "responses": [
      {
        "is": {
           "body": "..."
        }
      }
    ]
  }
]
```

It won't work if it has a header.

```
"defaultResponse": {
  "headers": {
      "Access-Control-Allow-Origin": "http://localhost:8000"
  }
},
"stubs": [
  {
    "responses": [
      {
        "is": {
           "headers": {},    // empty headers or not
           "body": "..."
        }
      }
    ]
  }
]
```

I think it would be great if it will a deep merge the `defaultResponse`.

Thank you for your response.

**Reply by paolooo** [reply]
I'll try to add these feature if this is okay?

**Comment by bbyars** [comment]
You bet! I’m actually surprised; I thought it already worked that way.
